### ISSUE TITLE
BUG: count_model Results cannot be pickled

### ISSUE THREAD

**Issue by jbrockmendel** [statsmodels/statsmodels#5677]
Setup based off test_count_model.TestZeroInflatedModel_logit.setup_class:

```
import statsmodels.api as sm
from six import BytesIO

data = sm.datasets.randhie.load(as_pandas=False)
exog = sm.add_constant(data.exog[:,1:4], prepend=False)
exog_infl = sm.add_constant(data.exog[:,0], prepend=False)
model = sm.ZeroInflatedPoisson(data.endog, exog, exog_infl=exog_infl, inflation='logit')
res = model.fit(method='newton', maxiter=500, disp=0)

c = BytesIO()
>>> res.save(c)
cPickle.PicklingError: Can't pickle <type 'instancemethod'>: attribute lookup __builtin__.instancemethod failed
```

The problem is the `results_class` attributes being set in `__init__`.  I'll be pushing a PR that fixes this shortly.

**Comment by bashtage** [comment]
Which python are you using?  I can't repro on 3.7.

**Comment by jbrockmendel** [comment]
That was with py27; it looks like this also is OK in py36.

**Comment by bashtage** [comment]
I suspect this is a Python 2.7 only problem.  Should probably just leave it be.

**Comment by bashtage** [comment]
This will not be fixed since Python 2.7 is no longer a target.
